packages/graphql/src/registry.ts

```diff
@@ -39,61 +50,148 @@ interface TSPTypeContext {
  *    by using thunks for fields/arguments.
  */
 export class GraphQLTypeRegistry {
-  // Stores intermediate TSP type information, keyed by TSP type name.
-  // TODO: make this more of a seen set
-  private TSPTypeContextRegistry: Map<string, TSPTypeContext> = new Map();
+  // Global registry to prevent GraphQL type name collisions
+  static #globalNameRegistry = new Set<TypeKey>();
 
-  // Stores materialized GraphQL types, keyed by their GraphQL name.
-  private materializedGraphQLTypes: Map<string, GraphQLNamedType> = new Map();
+  // Type maps for different GraphQL types
+  #objectTypes: ObjectTypeMap;
```

Comment:
Do you think we'll want a construct that can return the mapping for any type?
e.g.
```typescript
#getTypeMap(tspType: Type): TypeMap<Type, Any> {};
```